lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -114,3 +114,45 @@ function gutenberg_override_default_rest_server() {
 	return 'Gutenberg_REST_Server';
 }
 add_filter( 'wp_rest_server_class', 'gutenberg_override_default_rest_server', 1 );
+
+/**
+ *
+ * This function is hooked to the `rest_prepare_comment` filter and is responsible for updating the comment metadata based on the data provided in the REST API request.
+ *
+ * It performs the following tasks:
+ * - Updates the `block_comment` metadata for the comment based on the `meta` field in the request.
+ * - Updates the `comment_type` and `comment_approved` fields for the comment based on the corresponding fields in the request.
+ * - Retrieves the author's avatar URLs and adds them to the response data.
+ * - Retrieves the `block_comment` metadata for the comment and adds it to the response data.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Comment       $comment  The comment object.
+ * @param WP_REST_Request  $request  The request object.
+ * @return WP_REST_Response The updated response object.
+ */
+if ( ! function_exists( 'update_comment_meta_from_rest_request_6_7' ) ) {
+	function update_comment_meta_from_rest_request_6_7( $response, $comment, $request ) {
+
+		if ( ! empty( $request['comment_type'] ) ) {
+			$comment_data = array(
+				'comment_ID'       => $comment->comment_ID,
+				'comment_type'     => $request['comment_type'],
+				'comment_approved' => isset( $request['comment_approved'] ) ? $request['comment_approved'] : 0,
+			);
+
+			wp_update_comment( $comment_data );
```

Comment:
In our "Add Comment" REST API call, we pass a custom **comment_type** and utilize the **comment_approved** parameter to set the comment status to "hold." By default, the REST API assigns an approved status to comments submitted by registered WordPress users, but we've customized it to handle moderation based on our needs.

Reply:
Got it 👍 

You might want to extend the `admin_comment_types_dropdown` filter as well. This will eventually allow users to filter in the admin interface by the new `comment_type` that you're introducing. Unless you want to intentionally hide those comments from the interface.

Just to be sure I understand - why are we setting the comment status to `hold`? Isn't it enough to utilize the custom comment type?